p2p/src/peers.rs

```diff
@@ -83,24 +83,38 @@ impl Peers {
 	// Update the dandelion relay
 	pub fn update_dandelion_relay(&self) {
 		let peers = self.outgoing_connected_peers();
-
-		match thread_rng().choose(&peers) {
-			Some(peer) => {
-				// Clear the map and add new relay
-				let dandelion_relay = &self.dandelion_relay;
-				dandelion_relay.write().clear();
-				dandelion_relay
-					.write()
-					.insert(Utc::now().timestamp(), peer.clone());
-				debug!(
-					"Successfully updated Dandelion relay to: {}",
-					peer.info.addr
-				);
+		match &self.config.dandelion_peer {
```

Comment:
This would likely look nicer (and more idiomatic) if you first looked for and assigned the peer your want to set as relay, and then did the set on that. Look for the `find` method from `Iterator`.